src/libYARP_dev/src/devices/ControlBoardWrapper/ControlBoardWrapper.cpp

```diff
@@ -827,7 +827,7 @@ bool ControlBoardWrapper::detachAll()
 
         int devices=device.subdevices.size();
         for(int k=0;k<devices;k++)
-            device.getSubdevice(k)->detach();
+            if (device.getSubdevice(k)) device.getSubdevice(k)->detach();
```

Comment:
Please save the result of `device.getSubdevice(k)` in a variable instead of calling it twice